include/ace/generic/main.h

```diff
@@ -41,6 +41,8 @@ void __stack_chk_fail(void) {
 }
 #endif
 
+tStateManager *g_pGameStateManager;
```

Comment:
I have a problem with this and don't know how to solve it.

- Suppose someone wants to create files for state A and state B
- main.c includes generic/main.h and pushes state to A
- a.c  includes generic/main.h and pushes state to B
- then you have redeclaration of g_pGameStateManager

One has to create header extern for that variable. This is not documented, nor intuitive for beginners. So either it should be very clearly documented in docs *and* main.h or done in some other manner - dunno how though.

Reply:
That's why I removed default state manager... And I thing that is more correct way in this case.

Reply:
Yes and no. Yes because it solves multiple definition thingy, no because generic main was supposed to remove boilerplate used in 95% of cases, and gamestate manager is part of it.

I don't have a better idea how to solve it so yeah, you can remove it. Be sure to fix the docs/tutorial though. Also, while you're at it, refactor showcase code to runnable state.